### PR TITLE
Don't allocate intermediate array in Classify.extract_value

### PR DESCRIPTION
**Pull request by srt32**
This change removes the intermediate array made by `map`. It also allows us to loop only once. This change removes some allocations but maintains the same behavior.

### CODE REVIEW

File: test/benchmarks/bench_classify.rb [hunk 1]
```diff
@@ -51,7 +51,7 @@ def bench_allocations_with_cache
     Primer::Classify::Cache.preload!
     Primer::Classify.call(**@values)
 
-    assert_allocations 40 do
+    assert_allocations 39 do
```
Comment (srt32) [inline test/benchmarks/bench_classify.rb:54]:
```suggestion
    assert_allocations 38 do
```

File: test/benchmarks/bench_classify.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def bench_allocations_without_cache
     Primer::Classify::Cache.clear!
     Primer::Classify.call(**@values)
 
-    assert_allocations 115 do
+    assert_allocations 114 do
```
Comment (srt32) [inline test/benchmarks/bench_classify.rb:43]:
```suggestion
    assert_allocations 115 do
```

Comment (srt32) [inline test/benchmarks/bench_classify.rb:43]:
```suggestion
    assert_allocations 113 do
```

File: test/benchmarks/bench_classify.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ def bench_allocations_without_cache
     Primer::Classify::Cache.clear!
     Primer::Classify.call(**@values)
 
-    assert_allocations 115 do
+    assert_allocations 113 do
```
Comment (srt32) [inline test/benchmarks/bench_classify.rb:43]: This number seems to move around in CI...

**Review by joelhawksley — APPROVED** [review]

**Pull request merged**